### DISCUSSION TITLE
[Help]: Change snap points when using scrollTo method

### DISCUSSION THREAD

**Discussion by AndrewPJohnson** [davidjerleke/embla-carousel#873] [Help]
### Summary

I have a horizontal calendar carousel I just built using embla. When a user clicks on a calendar item I scroll it to the start of the carousel. The only way I can get that to work is by enabling dragFree. The problem is I like the animation when slidesToScroll is set to auto and dragFree is disabled (looks better in mobile browsers). When I set that to auto, the scrollTo method doesn't work like it does when dragFree is enabled. It scrolls to the nearest snap point using the number of slides to scroll instead of scrolling right to the clicked item.

I added a few examples in the CodeSandbox of what is happening to clear confusion. 

So is there a way to accomplish sliding an item to the start while also using slidesToScroll set to auto? My thoughts are disabling slidesToScroll and then enabling it after the scroll finishes. Or somehow changing the snap points.

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### CodeSandbox example

[CodeSandbox](https://codesandbox.io/p/sandbox/mutable-star-94rq5d?file=%2Fsrc%2FApp.js%3A8%2C53&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clw9v2jrc00063b6jlx4975yc%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clw9v2jrc00023b6jaqf3c0hg%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clw9v2jrc00033b6jyefl8ikl%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clw9v2jrc00053b6j0muqm247%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clw9v2jrc00023b6jaqf3c0hg%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clw9v2jrc00013b6jvir55z1z%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clw9v4jiq004h3b6j5viegja2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clw9woqu900023b6j206bqoui%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A8%252C%2522startColumn%2522%253A53%252C%2522endLineNumber%2522%253A8%252C%2522endColumn%2522%253A53%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clw9v2jrc00023b6jaqf3c0hg%2522%252C%2522activeTabId%2522%253A%2522clw9woqu900023b6j206bqoui%2522%257D%252C%2522clw9v2jrc00053b6j0muqm247%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clw9v2jrc00043b6j3w1chik1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clw9v2jrc00053b6j0muqm247%2522%252C%2522activeTabId%2522%253A%2522clw9v2jrc00043b6j3w1chik1%2522%257D%252C%2522clw9v2jrc00033b6jyefl8ikl%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clw9v2jrc00033b6jyefl8ikl%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by davidjerleke** [comment]
Hi @AndrewPJohnson,

I'm a bit confused. What should happen when you click on a slide? Should the carousel scroll to the snap point that slide belongs to?

**Reply by davidjerleke** [reply]
@AndrewPJohnson I will delete this issue if you don't answer soon.

**Reply by AndrewPJohnson** [reply]
I think it should scroll to the slide and update the snap points to now place that slide at the beginning, but maybe that's an unusual behavior.

**Reply by davidjerleke** [reply]
@AndrewPJohnson the behaviour you suggest would raise a lot of questions:

- How should `canScrollNext()` and `canScrollPrev()` work? Should they take snaps or slides into account?
- How should `scrollNext()` and `scrollPrev` work? Should it scroll to the next slide or snap?

As far as I see, if you set `slidesToScroll` to `auto`, you want slides to be grouped. It would get quite messy to eat the cake and have it too. However, if you just want to scroll to the snap that holds the slide you want to show, you can do this:

```tsx
function scrollToSnapBySlideIndex(slideIndex) {
  const { slideRegistry } = emblaApi.internalEngine();
  const snapIndex = slideRegistry.findIndex((snap) => snap.includes(slideIndex));

  if (typeof snapIndex === 'number') emblaApi.scrollTo(snapIndex);
}
```
